### ISSUE TITLE
WebKit doubles Keystrokes

### ISSUE THREAD

**Issue by N1k145** [eclipse-platform/eclipse.platform.swt#1200]
**Describe the bug**
When having the Focus in a WebKit Browser, keybindings are triggered twice.

I did a bit of Debugging and notices that there are coming two Events from the Display Que, but I was not able to determine how the second event ended up on the Que.

**To Reproduce**
A standalone SWT snippet to reproduce the behavior (preferable):
```
import javax.annotation.PostConstruct;

import org.eclipse.swt.SWT;
import org.eclipse.swt.browser.Browser;
import org.eclipse.swt.layout.GridData;
import org.eclipse.swt.widgets.Composite;

public class BrowserPart {

   @PostConstruct
   public void build(final Composite parent) {
      Browser browser = new Browser(parent, SWT.NONE);
      browser.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true));

      browser.setUrl("https://google.com");
   }
}
```
Alternatively, open the "Internal Browser" View in Eclipse

Click Ctrl + H, the Search Dialog opens
Click Esc or Cancle 
The Dialog closes
The Dialog opens again

**Expected behavior**
The Dialog opens once

**Screenshots**

https://github.com/eclipse-platform/eclipse.platform.swt/assets/15220836/a4ff296f-3bae-40e1-a393-e13020d35697

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [ ] Windows
- - [x] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
Ubuntu 22.4 X11

3. JRE/JDK version
 Java 17 Coretto

**Version since**
At least 23-09 (4.29)

**Workaround (or) Additional context**
Any known workarounds for the problem?

No

**Comment by N1k145** [comment]
@jukzi thank you for the hint. But I have to say that I have no idea how to tackle this. As this probably involves the native SWT/GTK parts from what I have seen during my initial investigation.

**Comment by akurtakov** [comment]
Is this still reproducible with Eclipse 2026-06?

**Comment by N1k145** [comment]
@akurtakov I am not able to reproduce the issue with 26-06 using the internal webbrowser. But I am able to reproduce it with 23-09. So I would asume its not related to my system and the issue is fixed in eclipse

**Comment by akurtakov** [comment]
23-09 is really ancient and so many things changes in swt. As you can't reproduce with 26-06 I'm closing this one.

**Comment by N1k145** [comment]
Yes it is, but it was the version I referenced in the issue. I just wanted to make sure that it's not an external factor. Im totally agreeing on closing the issue
